templates/app-ts/.vscode/launch.json

```diff
@@ -0,0 +1,19 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Fastify",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceRoot}/node_modules/.bin/fastify",
+            "args": ["start", "-l", "info", "./dist/app.js"],
```

Comment:
question: this may be a silly question. is it correct that the launch command runs on the dist folder, which I assume is the transpiled version of the app? Should it not execute on the original sources and invoke the same npm script?

Reply:
This is the correct usage when sourcemaps are enabled, that are referencing the original ts files.
This allows placing breakpoints on ts files:

![Screenshot 2021-11-11 at 16 22 52](https://user-images.githubusercontent.com/630796/141323576-63269802-cc82-4fd1-bf5c-3fef54b0e9d6.png)

I don't know inside out the implementation that allows this integration, but it works and it is a similar concept that applies also when debugging a frontend bundle with sourcemaps.

I know that if we add [ts-node](https://www.npmjs.com/package/ts-node) as a dependency we can achieve what you're pointing out.
I preferred to leave this dependency out because it starts to get opinionated, but it could be brought in if anyone has a strong point for going this way.